tests/e2e/test_spyre_prompt_logprobs.py

```diff
@@ -61,6 +61,31 @@ def test_prompt_logprobs(
     force_engine_shutdown(llm)
 
 
+@pytest.mark.parametrize("backend", get_spyre_backend_list())
+@pytest.mark.parametrize("model", get_spyre_model_list())
+def test_prompt_logprobs_work_with_logprobs(
+    backend: str,
+    model: str,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    '''
+    This test checks that prompt_logprobs and logprobs can work together.
+    '''
+
+    prompts = get_chicken_soup_prompts(4)
+
+    monkeypatch.setenv("VLLM_SPYRE_DYNAMO_BACKEND", backend)
+    monkeypatch.setenv("VLLM_SPYRE_ENABLE_PROMPT_LOGPROBS", "1")
+    llm = LLM(model, tokenizer=model)
+
+    responses: list[RequestOutput] = llm.generate(
+        prompts, sampling_params=SamplingParams(prompt_logprobs=5, logprobs=5))
+
+    for r in responses:
+        assert r.outputs[0].logprobs is not None
+    force_engine_shutdown(llm)
```

Comment:
can we add some comments on when we have to use this function? Also I wonder if we can make it such that it's part of pytest teardown automatically in case someone forgets to add this which would result in the weird addr-in-use error